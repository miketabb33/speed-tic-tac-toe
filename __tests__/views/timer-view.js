import React from 'react'
import renderer from 'react-test-renderer'
import TimerView from '../../views/timer-view'

test('render x active', () => {
	const subject = renderer.create(
		<TimerView player={'X'} activePlayer={'X'} remainingTime={5} />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render x inactive', () => {
	const subject = renderer.create(
		<TimerView player={'X'} activePlayer={'O'} remainingTime={5}/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render o active', () => {
	const subject = renderer.create(
		<TimerView player={'O'} activePlayer={'O'} remainingTime={5}/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render o inactive', () => {
	const subject = renderer.create(
		<TimerView player={'O'} activePlayer={'X'} remainingTime={5}/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})