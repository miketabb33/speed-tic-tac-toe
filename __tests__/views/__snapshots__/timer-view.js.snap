// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render o active 1`] = `
<div
  className="timerView timerViewActiveO"
>
  <div
    className="timerViewImage"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="O player marker"
        data-nimg="intrinsic"
        decoding="async"
        draggable="false"
        src="/_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75"
        srcSet="/_next/image?url=%2Fplayer-markers%2FO.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="O player marker"
          data-nimg="intrinsic"
          decoding="async"
          draggable="false"
          loading="lazy"
          src="/_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fplayer-markers%2FO.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <div
    className="timerViewText timerViewO "
  >
    00:05
  </div>
</div>
`;

exports[`render o inactive 1`] = `
<div
  className="timerView timerViewInactive"
>
  <div
    className="timerViewImage"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="O player marker"
        data-nimg="intrinsic"
        decoding="async"
        draggable="false"
        src="/_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75"
        srcSet="/_next/image?url=%2Fplayer-markers%2FO.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="O player marker"
          data-nimg="intrinsic"
          decoding="async"
          draggable="false"
          loading="lazy"
          src="/_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fplayer-markers%2FO.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FO.png&w=64&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <div
    className="timerViewText timerViewO "
  >
    00:05
  </div>
</div>
`;

exports[`render x active 1`] = `
<div
  className="timerView timerViewActiveX"
>
  <div
    className="timerViewImage"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="X player marker"
        data-nimg="intrinsic"
        decoding="async"
        draggable="false"
        src="/_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75"
        srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="X player marker"
          data-nimg="intrinsic"
          decoding="async"
          draggable="false"
          loading="lazy"
          src="/_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <div
    className="timerViewText timerViewX "
  >
    00:05
  </div>
</div>
`;

exports[`render x inactive 1`] = `
<div
  className="timerView timerViewInactive"
>
  <div
    className="timerViewImage"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="X player marker"
        data-nimg="intrinsic"
        decoding="async"
        draggable="false"
        src="/_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75"
        srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="X player marker"
          data-nimg="intrinsic"
          decoding="async"
          draggable="false"
          loading="lazy"
          src="/_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fplayer-markers%2FX.png&w=32&q=75 1x, /_next/image?url=%2Fplayer-markers%2FX.png&w=64&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <div
    className="timerViewText timerViewX "
  >
    00:05
  </div>
</div>
`;
